server.py

```diff
@@ -62,7 +64,32 @@ def run_tasks(services):
         logging.info('All systems ready.')
         loop.run_forever()
     except KeyboardInterrupt:
-        loop.run_until_complete(services.get('app_svc').teardown())
+        loop.run_until_complete(services.get('app_svc').teardown(main_config_file=args.environment))
+
+
+def make_secure_config(config_name):
+    logging.info('Generating new {}.yml configuration file'.format(args.environment))
+    with open('conf/default.yml', 'r') as fle:
+        config = yaml.safe_load(fle)
+    secret_options = ('api_key_blue', 'api_key_red', 'crypt_salt', 'encryption_key')
+    for option in secret_options:
+        config[option] = secrets.token_urlsafe()
+
+    config['users'] = dict(red=dict(red=secrets.token_urlsafe()),
+                           blue=dict(blue=secrets.token_urlsafe()))
+
+    config_path = 'conf/{}.yml'.format(args.environment)
+    with open(config_path, 'w') as fle:
+        yaml.dump(config, fle, default_flow_style=False)
+
+    msg = "\nGenerated new config with random secrets:\n"
+    for option in secret_options:
+        msg += "\t{}={}\n".format(option, config[option])
```

Comment:
This messaging could be a bit cleaner. I think we could strip out the \n and \t

Reply:
I decided that a jinja template would be prettier and be a nicer way to extract nested values https://github.com/mitre/caldera/pull/1544/files#diff-6449db3e452640506e59349e927485e8R9